File: Semestrul 3/MAP/lab 10/tasks/SortingTask.cs
namespace lab10;

public class SortingTask : Task
{
    private List<int> _numbers;
    private AbstractSorter _sorter;

    public SortingTask(string taskId, string description, List<int> numbers)
        : base(taskId, description)
    {
        this._numbers = numbers;
        this._sorter = new DefaultSorter(numbers);
    }

    public SortingTask(string taskId, string description, List<int> numbers, SortingStrategy method)
        : base(taskId, description)
    {
        this._numbers = numbers;

        this._sorter = method switch
        {
            SortingStrategy.Quick => new QuickSorter(numbers),
            SortingStrategy.Bubble => new BubbleSorter(numbers),
            _ => new DefaultSorter(numbers)
        };
    }

    public override void Execute()
    {
        _sorter.Sort();

        foreach (var number in _numbers)
        {
            Console.Write(number + " ");
        }
        Console.WriteLine();
    }
}

File: Semestrul 3/MAP/lab 02/src/Tasks/SortingTask.java
package Tasks;

import Sorters.AbstractSorter;
import Sorters.BubbleSorter;
import Sorters.DefaultSorter;
import Sorters.QuickSorter;
import Sorters.SortingMethod;
import java.util.List;

public class SortingTask extends Task{
    List<Integer> numbers;
    AbstractSorter sorter;
    public SortingTask(String taskID, String description, List<Integer> nr) {
        super(taskID, description);
//        this.numbers = new List<Integer>(nr); ?? cum sa fac sa mearga
        this.numbers = nr;
        sorter = new DefaultSorter(numbers);
    }
    public SortingTask(String taskID, String description, List<Integer> nr, SortingMethod method) {
        super(taskID, description);
        this.numbers = nr;
        sorter = new DefaultSorter(numbers);
        switch (method){
            case QUICK -> sorter = new QuickSorter(numbers);
            case BUBBLE -> sorter = new BubbleSorter(numbers);
        }
    }
    @Override
    public void execute() {
        sorter.sort();
        for(var number: numbers){
            System.out.print(number + " ");
        }
        System.out.println();
    }

}
